server.js

```diff
@@ -7476,35 +7478,45 @@ cache({
 
     var badgeData = getBadgeData('custom badge', query);
 
-    if (!query.uri){
+    if (!query.uri || !query.query){
       setBadgeColor(badgeData, 'red');
-      badgeData.text[1] = 'no uri specified';
+      badgeData.text[1] = !query.uri ? 'no uri specified' : 'no query specified';
       sendBadge(format, badgeData);
       return;
     }
     var uri = encodeURI(decodeURIComponent(query.uri));
 
     request(uri, (err, res, data) => {
       try {
-        if (res && res.statusCode === 404)
-          throw 'invalid resource';
-
-        if (err != null || !res || res.statusCode !== 200)
-          throw 'inaccessible';
+        if (checkErrorResponse(badgeData, err, res, 'uri not found')) {
```

Comment:
Updated to use `checkErrorResponse()` but not sure if `uri not found || not found` is the best message for a 404 error on this badge?

Reply:
Maybe `file not found` or `resource not found`?